erpnext/accounts/doctype/bank_reconciliation/bank_reconciliation.py

```diff
@@ -90,10 +90,82 @@ def update_clearance_date(self):
 					where name=%s""".format(d.payment_document), 
 				(d.clearance_date, nowdate(), d.payment_entry))
 				
+				cheque_status = "Done"
+				frappe.db.sql("update `tabPayment Entry` set cheque_status=%s where clearance_date=%s",
+							(cheque_status, d.clearance_date))
 				clearance_date_updated = True
 
 		if clearance_date_updated:
 			self.get_payment_entries()
 			msgprint(_("Clearance Date updated"))
 		else:
 			msgprint(_("Clearance Date not mentioned"))
+	
+	
+	def update_return_date(self):
+		return_date_updated = False
+		for d in self.get('payment_entries'):
+			if d.return_date:
+				if not d.payment_document:
+					frappe.throw(_("Row #{0}: Payment document is required to complete the trasaction"))
+				
+				if d.cheque_date and getdate(d.return_date) < getdate(d.cheque_date):
+					frappe.throw(_("Row #{0}: Return date {1} cannot be before Cheque Date {2}")
+						.format(d.idx, d.return_date, d.cheque_date))
+				
+			if d.return_date or self.include_reconciled_entries:
+				if not d.return_date:
```

Comment:
what is the use of this?

Reply:
We are updating the status of payment entry by updating return date to related payment entry.